spec/forms/base_form_spec.rb

```diff
@@ -84,4 +84,24 @@
       end
     end
   end
+
+  context 'with non-existing attributes' do
+    let(:object) { create :user, email: 'asya@purple-magic.com' }
+    subject { described_class.new(object) }
+
+    describe '#submit' do
+      let(:params) { { password: '123456' } }
+
+      it 'does not update non-existing attributes' do
+        expect(object).to receive(:save).and_return(true)
+        expect(object).to receive(:reload)
+
+        previous_email = object.email
+
+        subject.submit(params)
+
+        expect(object.email).to eq previous_email
```

Comment:
could be better to move email value to separate let (like let(:user_email) { "..." }),  and use it here instead of previous_email. 
To mitigate risk that the test could pass if both previous_email and object.email are nil

Reply:
yes